tools/tool-testing/selftest.js

```diff
@@ -2121,7 +585,7 @@ export function runTests(options) {
     } catch (e) {
       failure = e;
     } finally {
-      runningTest = null;
+      setRunningTest(null);
       test.cleanup();
     }
```

Comment:
What if you moved this code into a `static` method of the `Run` class, and here wrote something like
```js
const {
  error,
  durationMs,
} = Run.runTest(test, parseStackMarkBottom(() => {
  test.f(options);
});
```
That way you wouldn't have to be careful about calling `setRunningTest` twice, and the code that needs to know the current test can be (more) responsible for managing that information.